### ISSUE TITLE
File deletion on server not reflected in local after _clasp pull_

### ISSUE THREAD

**Issue by tanyagupta** [google/clasp#291]
(Please search [existing issues](https://github.com/google/clasp/issues) before creating a new one.)

## Expected Behavior
_clasp pull_ should pull latest changes made remotely both additions and deletions (including added or deleted files)

## Actual Behavior
Using _clasp pull_ I am finding files deleted in the server are not being deleted locally.

## Steps to Reproduce the Problem

  1. clasp clone <scriptId>
  1. Then delete a file on the server
  1. clasp pull 
  1. local copy still shows deleted file 

## Specifications

  - Node version (`node -v`): 8.9.4
  - Version (`clasp -v`): 1.5.2
  - OS (Mac/Linux/Windows): Mac

**Comment by grant** [comment]
Yes, this is a known issue. The API doesn't make it easy to safely delete files.

I guess I could delete all non-ignored, non-pulled files when you pull. This would still not delete ignored files though.

**Comment by tanyagupta** [comment]
Thanks @grant good to know.

**Comment by grant** [comment]
In reality, we'd need a better file system with API like git to detect server deletions.

**Comment by Anuvad** [comment]
Any update on this please?

**Comment by coreyjansen** [comment]
+1 to many times i have duplicated a file because of this

**Comment by sqrrrl** [comment]
Very hesitant to delete anything local due to the severity of consequences when it's wrong.

Possible to do *if* clasp tracks previously pushed items and knows for certain that a file was tracked. Possible, but need to make sure it isn't error prone and results in accidental deletions if somebody clones/pulls in the wrong location.

**Comment by TomerFi** [comment]
Any updates on this?

**Comment by sqrrrl** [comment]
No plans at the moment.

You can roughly accomplish this with a wrapper script or alias to delete source files ahead of pulling (particularly if you're using git locally too to minimize risk) but again, one of those things that is dangerous and needs a lot of thought. But if you're willing to take that risk that'll get you most of the way there :)

**Comment by sqrrrl** [comment]
Alright... convinced myself to add it :)

Next release will have a -d option to do this. It should only consider files that *would have* been pushed and honor .claspignore and all that. Will prompt for each file before deleting, can use -f flag to force.
